group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2904,6 +2908,135 @@ private void validateGenericGroupHeartbeat(
         }
     }
 
+    /**
+     * Handle a generic group LeaveGroup request.
+     *
+     * @param context        The request context.
+     * @param request        The actual LeaveGroup request.
+     *
+     * @return The LeaveGroup response and the GroupMetadata record to append if the group
+     *         no longer has any member.
+     */
+    public CoordinatorResult<LeaveGroupResponseData, Record> genericGroupLeave(
+        RequestContext context,
+        LeaveGroupRequestData request
+    ) throws UnknownMemberIdException, GroupIdNotFoundException {
+        GenericGroup group = getOrMaybeCreateGenericGroup(request.groupId(), false);
+        if (group.isInState(DEAD)) {
+            return new CoordinatorResult<>(Collections.emptyList(),
+                new LeaveGroupResponseData()
+                    .setErrorCode(COORDINATOR_NOT_AVAILABLE.code())
+            );
+        }
+
+        CoordinatorResult<Void, Record> coordinatorResult = EMPTY_RESULT;
+        List<MemberResponse> memberResponses = new ArrayList<>();
+
+            for (MemberIdentity member : request.members()) {
+                // The LeaveGroup API allows administrative removal of members by GroupInstanceId
+                // in which case we expect the MemberId to be undefined.
+                if (member.memberId().equals(UNKNOWN_MEMBER_ID)) {
```

Comment:
nit: Would it make sense to use `UNKNOWN_MEMBER_ID.equals(member.memberId())`? That would make the code a little more robust in the case where `memberId` would be null.